Sources/ArcGISToolkit/Components/Offline/OfflineManager.swift

```diff
@@ -73,6 +80,49 @@ class OfflineManager {
             jobCompletionAction?(job)
         }
     }
+    
+    /// Saves the portal item to UserDefaults.
+    /// - Parameters:
+    ///   - id: The portal item ID.
+    ///   - itemJSON: The portal item JSON.
+    func savePortalItem(_ portalItemID: PortalItem.ID, itemJSON: String) {
+        var savedMapIDs = UserDefaults.standard.stringArray(forKey: "offline") ?? []
+        
+        let id = portalItemID.description
+        
+        if !savedMapIDs.contains(id) {
+            savedMapIDs.append(id)
+        }
+        
+        // Save portal item ID.
+        UserDefaults.standard.set(savedMapIDs, forKey: "offline")
+        
+        // Save portal item JSON.
+        UserDefaults.standard.set(itemJSON, forKey: id)
+        
+        loadMaps()
```

Comment:
You don't need to load the maps from the defaults again. That is re-creating portal items and portals. You just need to save to defaults, then modify the in-memory collection.